tools/repo-tool

```diff
@@ -95,18 +109,27 @@ upload() {
 }
 
 build() {
+  [ -n "$3" ] && DEVICE="$3" || DEVICE=""
+
   for PROJECT in $2; do
-    for archfile in projects/$PROJECT/linux/linux.*.conf ; do
-      ARCH=`echo $archfile | sed -n '$s/\.conf//;$s/.*\.//p'`
-      for package in $(find $1 -iname package.mk) ; do
-        (
-        . $package
-        if [ "$PKG_IS_ADDON" = "yes" ] ; then
-          ADDON=$PKG_NAME
-          PROJECT=$PROJECT ARCH=$ARCH ./scripts/create_addon $ADDON
-        fi
-        )
+    if [ -n "$DEVICE" ]; then
+      for archfile in projects/$PROJECT/devices/$DEVICE/linux/linux.*.conf; do
+        ARCH=`echo $archfile | sed -n '$s/\.conf//;$s/.*\.//p'`
       done
+    else
+      for archfile in projects/$PROJECT/linux/linux.*.conf; do
+        ARCH=`echo $archfile | sed -n '$s/\.conf//;$s/.*\.//p'`
```

Comment:
There's a globbing issue here when `projects/$PROJECT/linux/linux.*.conf ` doesn't expand into a valid filename - you might end up with `ARCH=*`?. I think this will happen when the script is run for all projects as project=RPi won't have a device and thescript won't find a valid config. Adding some protecting wild be advisable:
```
  [ -f $archfile ] && ARCH=`echo $archfile | sed -n '$s/\.conf//;$s/.*\.//p'` || ARCH=
```

Same goes for line 117, really. And I'm not sure what, if anything, to do about the project=RPi without a device situation - only thing I can think of is another loop iterating over devices directories...